File: SDKs/googleapis.com/monitoring/v1/go/pkg/models/operations/monitoringprojectslocationprometheusapiv1labelvalues.go
package operations

import (
"openapi/pkg/models/shared")

type MonitoringProjectsLocationPrometheusAPIV1LabelValuesPathParams struct {
    Label string `pathParam:"style=simple,explode=false,name=label"`
    Location string `pathParam:"style=simple,explode=false,name=location"`
    Name string `pathParam:"style=simple,explode=false,name=name"`
    
}

type MonitoringProjectsLocationPrometheusAPIV1LabelValuesQueryParams struct {
    DollarXgafv *shared.XgafvEnum `queryParam:"style=form,explode=true,name=$.xgafv"`
    AccessToken *string `queryParam:"style=form,explode=true,name=access_token"`
    Alt *shared.AltEnum `queryParam:"style=form,explode=true,name=alt"`
    Callback *string `queryParam:"style=form,explode=true,name=callback"`
    End *string `queryParam:"style=form,explode=true,name=end"`
    Fields *string `queryParam:"style=form,explode=true,name=fields"`
    Key *string `queryParam:"style=form,explode=true,name=key"`
    Match *string `queryParam:"style=form,explode=true,name=match"`
    OauthToken *string `queryParam:"style=form,explode=true,name=oauth_token"`
    PrettyPrint *bool `queryParam:"style=form,explode=true,name=prettyPrint"`
    QuotaUser *string `queryParam:"style=form,explode=true,name=quotaUser"`
    Start *string `queryParam:"style=form,explode=true,name=start"`
    UploadType *string `queryParam:"style=form,explode=true,name=uploadType"`
    UploadProtocol *string `queryParam:"style=form,explode=true,name=upload_protocol"`
    
}

type MonitoringProjectsLocationPrometheusAPIV1LabelValuesSecurityOption1 struct {
    Oauth2 shared.SchemeOauth2 `security:"scheme,type=oauth2"`
    Oauth2c shared.SchemeOauth2c `security:"scheme,type=oauth2"`
    
}

type MonitoringProjectsLocationPrometheusAPIV1LabelValuesSecurityOption2 struct {
    Oauth2 shared.SchemeOauth2 `security:"scheme,type=oauth2"`
    Oauth2c shared.SchemeOauth2c `security:"scheme,type=oauth2"`
    
}

type MonitoringProjectsLocationPrometheusAPIV1LabelValuesSecurityOption3 struct {
    Oauth2 shared.SchemeOauth2 `security:"scheme,type=oauth2"`
    Oauth2c shared.SchemeOauth2c `security:"scheme,type=oauth2"`
    
}

type MonitoringProjectsLocationPrometheusAPIV1LabelValuesSecurity struct {
    Option1 *MonitoringProjectsLocationPrometheusAPIV1LabelValuesSecurityOption1 `security:"option"`
    Option2 *MonitoringProjectsLocationPrometheusAPIV1LabelValuesSecurityOption2 `security:"option"`
    Option3 *MonitoringProjectsLocationPrometheusAPIV1LabelValuesSecurityOption3 `security:"option"`
    
}

type MonitoringProjectsLocationPrometheusAPIV1LabelValuesRequest struct {
    PathParams MonitoringProjectsLocationPrometheusAPIV1LabelValuesPathParams 
    QueryParams MonitoringProjectsLocationPrometheusAPIV1LabelValuesQueryParams 
    Security MonitoringProjectsLocationPrometheusAPIV1LabelValuesSecurity 
    
}

type MonitoringProjectsLocationPrometheusAPIV1LabelValuesResponse struct {
    ContentType string 
    HTTPBody *shared.HTTPBody 
    StatusCode int64 
    
}

File: SDKs/googleapis.com/monitoring/v1/typescript/src/sdk/models/operations/monitoringprojectslocationprometheusapiv1labelvalues.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class MonitoringProjectsLocationPrometheusApiV1LabelValuesPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=label" })
  label: string;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=location" })
  location: string;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=name" })
  name: string;
}


export class MonitoringProjectsLocationPrometheusApiV1LabelValuesQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=$.xgafv" })
  dollarXgafv?: shared.XgafvEnum;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=access_token" })
  accessToken?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=alt" })
  alt?: shared.AltEnum;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=callback" })
  callback?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=end" })
  end?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=fields" })
  fields?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=key" })
  key?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=match" })
  match?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=oauth_token" })
  oauthToken?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=prettyPrint" })
  prettyPrint?: boolean;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=quotaUser" })
  quotaUser?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=start" })
  start?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=uploadType" })
  uploadType?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=upload_protocol" })
  uploadProtocol?: string;
}


export class MonitoringProjectsLocationPrometheusApiV1LabelValuesSecurityOption1 extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "security, scheme=true;type=oauth2" })
  oauth2: shared.SchemeOauth2;

  @SpeakeasyMetadata({ data: "security, scheme=true;type=oauth2" })
  oauth2c: shared.SchemeOauth2c;
}


export class MonitoringProjectsLocationPrometheusApiV1LabelValuesSecurityOption2 extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "security, scheme=true;type=oauth2" })
  oauth2: shared.SchemeOauth2;

  @SpeakeasyMetadata({ data: "security, scheme=true;type=oauth2" })
  oauth2c: shared.SchemeOauth2c;
}


export class MonitoringProjectsLocationPrometheusApiV1LabelValuesSecurityOption3 extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "security, scheme=true;type=oauth2" })
  oauth2: shared.SchemeOauth2;

  @SpeakeasyMetadata({ data: "security, scheme=true;type=oauth2" })
  oauth2c: shared.SchemeOauth2c;
}


export class MonitoringProjectsLocationPrometheusApiV1LabelValuesSecurity extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "security, option=true" })
  option1?: MonitoringProjectsLocationPrometheusApiV1LabelValuesSecurityOption1;

  @SpeakeasyMetadata({ data: "security, option=true" })
  option2?: MonitoringProjectsLocationPrometheusApiV1LabelValuesSecurityOption2;

  @SpeakeasyMetadata({ data: "security, option=true" })
  option3?: MonitoringProjectsLocationPrometheusApiV1LabelValuesSecurityOption3;
}


export class MonitoringProjectsLocationPrometheusApiV1LabelValuesRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: MonitoringProjectsLocationPrometheusApiV1LabelValuesPathParams;

  @SpeakeasyMetadata()
  queryParams: MonitoringProjectsLocationPrometheusApiV1LabelValuesQueryParams;

  @SpeakeasyMetadata()
  security: MonitoringProjectsLocationPrometheusApiV1LabelValuesSecurity;
}


export class MonitoringProjectsLocationPrometheusApiV1LabelValuesResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  httpBody?: shared.HttpBody;

  @SpeakeasyMetadata()
  statusCode: number;
}
